test/package_list_files_test.dart

```diff
@@ -135,8 +405,34 @@ void main() {
           (e) => e.message,
           'message',
           contains(
-            'Pub does not support publishing packages with '
-            'non-resolving symlink:',
+            'Could not resolve symbolic link',
+          ),
+        ),
+      ),
+    );
+  });
+
+  test('throws on loop file symlinks', () async {
+    await d.dir(appPath, [
+      d.pubspec({'name': 'myapp'}),
+      d.file('file1.txt', 'contents'),
+      d.file('file2.txt', 'contents'),
+      d.dir('subdir', [
+        d.dir('a', [d.file('file')]),
+      ]),
+    ]).create();
+    Link(p.join(d.sandbox, appPath, 'subdir', 'symlink')).createSync('symlink');
```

Comment:
Please consider using `await d.link(...).create();`

You won't have to use `d.sandbox` and it'll be in the same style as the others.

------

If it's a special reason to not use it here, that's fine.